op-node/rollup/derive/engine_controller.go

```diff
@@ -72,6 +80,10 @@ func (e *EngineController) Finalized() eth.L2BlockRef {
 	return e.finalizedHead
 }
 
+func (e *EngineController) BackupUnsafeL2Head() eth.L2BlockRef {
```

Comment:
Ideally we should not expose this backup unsafe L2 head.

Reply:
I agree, it makes sense to keep this internal

Reply:
While testing, I want to sanity check state of `backupUnsafeHead`. Are there any other methods to do this, while not exposing? Also, `pendingSafeHead` has a similar pattern. 

I have noticed that engine controller is refactored at https://github.com/ethereum-optimism/optimism/pull/8966. Currently it does not remove exposure of its internal variables, such as `pendingSafeHead`.

Reply:
If we could remove pendingSafeHead, I would. Tests inside the derive package will still be able to access this internal field.